lib/components/form/element/combobox.js

```diff
@@ -767,8 +767,7 @@ class Combobox extends React.Component {
 
                     // Don't include the disabled options that match the regex unless we specified we want them to show.
                     // If we want them to show then add them whether they match the regex or not.
-                    if ((!isDisabled && isMatch)
-                        || (isDisabled && this.props.showDisabledOptionsInResults)) {
+                    if ((!isDisabled && isMatch) || (isDisabled && this.props.showDisabledOptionsInResults && isMatch)) {
```

Comment:
The only difference besides style is that I added `&& isMatch` to the second condition because by the old logic, we would add every single non-match to matches lol

Reply:
Isn't this just the same as `if (isDisabled && isMatch)`?

`if (A || A & B)` is always going to be true if `A` is true, so it simplifies to `if (A)`.